### PR TITLE
Qa 0.21

### PR DESCRIPTION
**Pull request by Pavan-Gunda**
**What this PR does / why we need it**:
merge QA to release 
**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:

**Add a screenshot or an example to illustrate the proposed solution:**

**Checklist:**

- [ ] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [ ] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -44,6 +44,8 @@
 - fluentd are upgraded from 2.4.0 to 5.0.15 and fluentd-elasticsearch are upgraded from 10.2.1 to 13.3.0. Elastisys also made their own fluentd-elasticsearch container image using fluentd-plugin-opensearch 1.0.4 to work with opensearch.
 - Changed the grafana image tag from 8.2.7 to 8.4.7 in both user-grafana and kube-prometheus-stack, as the latter has less vulnerabilities.
 - Changed the harbor chartmuseum image tag from 2.2.1 to 2.4.2 as the latter has less vulnerabilities.
+- Exposed fluentd-elasticsearch buffer settings in the wc-config.yaml
+- Increased fluentd `FOR` alerts to 30m, decrease number of unnecessary alerts.
```
Comment (OlleLarsson) [inline CHANGELOG.md:48]: ?
```suggestion
- Increased fluentd `FOR` alerts to 30m, should decrease number of false-positive alerts.
```

File: migration/v0.20.x-v0.21.x/upgrade-apps.md [hunk 2]
```diff
@@ -4,11 +4,19 @@
...
+    This will take a backup into `backups/` before modifying any files.
 
-3. Upgrade applications:
+    ```bash
+    bin/ck8s init
+    ```
+
+2. Remove grafana-ops chart with all the dashboards: `migration/v0.20.x-v0.21.x/remove-grafana-ops.sh`
+
+3. Rollover all streams to make fluentd have all logging fields/mappings: `migration/v0.20.x-v0.21.x/data-stream-rollovers.sh`
```
Comment (OlleLarsson) [inline migration/v0.20.x-v0.21.x/upgrade-apps.md:17]:
```suggestion
@robinelastisys why is this nessecary? If nessecary why is it done before apps apply?
```

Comment (robinelastisys) [reply migration/v0.20.x-v0.21.x/upgrade-apps.md:17]: It is necessary as fluentd needs to have all fields mapped out in OpenSearch to have it work correctly. If not, fluentd will say it simply cannot map the log field.

Comment (OlleLarsson) [reply migration/v0.20.x-v0.21.x/upgrade-apps.md:17]: Okay, I just fail to see what the impact of rolling over before upgrading has, the same mappings will be present after the rollover when fluentd pushes logs to opensearch.

Comment (OlleLarsson) [reply migration/v0.20.x-v0.21.x/upgrade-apps.md:17]: @Pavan-Gunda or @viktor-f can you try an upgrade without running this?

File: migration/v0.20.x-v0.21.x/upgrade-apps.md [hunk 3]
```diff
@@ -4,11 +4,19 @@
 
 ## Steps
```
Comment (robinAwallace) [inline migration/v0.20.x-v0.21.x/upgrade-apps.md:6]: Can you add a step to run `get-requirements` to update kubectl.

File: migration/v0.20.x-v0.21.x/upgrade-apps.md (hunk 2 above)
Comment (Pavan-Gunda) [reply migration/v0.20.x-v0.21.x/upgrade-apps.md:17]: I will do that!!

File: migration/v0.20.x-v0.21.x/upgrade-apps.md (hunk 3 above)
Comment (Pavan-Gunda) [reply migration/v0.20.x-v0.21.x/upgrade-apps.md:10]: done

**Pull request merged**